File: packages/cli/test/fixtures/ranking-corpora/php/src/Handlers/CreateHandler.php
<?php

namespace RankingPhp\Handlers;

use RankingPhp\Util\Logger;

/** Validates and records a new user by name. */
class CreateHandler
{
    public function createUser(string $name): void
    {
        if ($name === '') {
            Logger::logError('createUser', new \InvalidArgumentException('name must not be empty'));
            return;
        }
        Logger::logInfo("created user {$name}");
    }
}

File: packages/cli/test/fixtures/ranking-corpora/csharp/CreateHandler.cs
namespace RankingCSharp;

/// <summary>Validates and records a new user by name.</summary>
public class CreateHandler
{
    public void CreateUser(string name)
    {
        if (string.IsNullOrEmpty(name))
        {
            Logger.LogError("CreateUser", new ArgumentException("name must not be empty"));
            return;
        }

        Logger.LogInfo($"created user {name}");
    }
}
